subprojects/core/src/main/java/org/gradle/initialization/DefaultSettings.java

```diff
@@ -275,4 +284,29 @@ public void buildCache(Action<? super BuildCacheConfiguration> action) {
     public BuildCacheConfigurationInternal getBuildCache() {
         throw new UnsupportedOperationException();
     }
+
+    @Override
+    public void pluginRepositories(Action<? super PluginRepositoriesSpec> pluginSettings) {
+        pluginSettings.execute(getPluginRepositoriesSpec());
+    }
+
+    @Override
+    public void pluginManagement(Action<? super PluginManagementSpec> rule) {
+        rule.execute(getPluginManagementSpec());
+    }
+
+    @Override
+    public PluginRepositoriesSpec getPluginRepositoriesSpec() {
+        Instantiator instantiator = services.get(Instantiator.class);
```

Comment:
I think we can get rid of this cruft now that we are in the `Settings` scope and use our normal service injection instead of creating a new instance every time. I.e. `PluginRepositoriesSpec` and `PluginManagementSpec` can live in the `SettingsScopedServices` and just be injected here, like the build cache configuration a few lines above.